src/package.ts

```diff
@@ -0,0 +1,57 @@
+import { privateEncrypt } from 'crypto';
+import contextKeys from './contextKeys';
+import { exec, pathExists } from './utilities';
+
+export class SPMPackage {
+	private constructor(
+        readonly folder: string,
+        public contents: any
+    ) {
+        this.setContextKeys()
+    }
+
+    public static async create(folder: string): Promise<SPMPackage> {
+        try {
+            let contents = await SPMPackage.loadPackage(folder)
+            return new SPMPackage(folder, contents)
+        } catch(error) {
+            // TODO: output errors
+            return new SPMPackage(folder, null)
+        }
+    }
+
+    public static async loadPackage(folder: string): Promise<any> {
```

Comment:
Does this have to be static? Can it be just `load`?

Reply:
It is being called from the factory function, so needs to be static

Reply:
You are right? And it is fixed